### PR TITLE
Feature/search bar improvements

### PR DESCRIPTION
**Pull request by learykara**

### CODE REVIEW

File: src/AppBar/NavLink.jsx [hunk 1]
```diff
@@ -51,6 +54,25 @@ class SearchLink extends React.Component {
...
+    componentDidMount() {
+        window.addEventListener('keydown', this._handleWindowKeyDown);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('keydown', this._handleWindowKeyDown);
+    }
+
+    _handleWindowKeyDown = (event) => {
+        if (event.which === SLASH_KEY_CODE) {
```
Comment (ripleyaffect) [inline src/AppBar/NavLink.jsx]: will this make the search bar pop up if typing in the raven editor?

File: src/AppBar/NavLink.jsx [hunk 2]
```diff
@@ -51,6 +54,25 @@ class SearchLink extends React.Component {
...
+            this.setState({open: true});
+            this.refs.searchBar._autoFocus();
```
Comment (ripleyaffect) [inline src/AppBar/NavLink.jsx]: you shouldn't be calling an "internal" method like this, from outside the component. you can expose it with a public facing method called `autoFocus` which just calls the internal one or you can remove the underscore at the beginning

File: src/AppBar/NavLink.jsx [hunk 3]
```diff
@@ -51,6 +54,25 @@ class SearchLink extends React.Component {
...
+        }
+        else if (event.which === ESC_KEY_CODE) {
+            this.setState({open: false});
+            this.refs.searchBar._unFocus();
```
Comment (ripleyaffect) [inline src/AppBar/NavLink.jsx]: ditto on the api

File: src/AppBar/NavLink.jsx [hunk 4]
```diff
@@ -66,7 +88,7 @@ class SearchLink extends React.Component {
 
         mobile ?
             window.location = `${config.projects.url}/search`
-        :   active ? this.refs.searchBar._autoFocus()
+        :   active ? this.refs.searchBar._wiggle()
```
Comment (ripleyaffect) [inline src/AppBar/NavLink.jsx]: tritto on the api

File: src/SearchBar/SearchBar.jsx [hunk 5]
```diff
@@ -54,7 +54,21 @@ class SearchBar extends React.Component {
...
+    React.findDOMNode(this.refs.input).value = this.state.searchTerm;
+  }
+
+  _unFocus = () => {
+    React.findDOMNode(this.refs.input).blur();
+  }
+
+  _wiggle = () => {
+    this._autoFocus();
+    React.findDOMNode(this.refs.input).classList.add('wiggle');
```
Comment (ripleyaffect) [inline src/SearchBar/SearchBar.jsx]: You shouldn't be adding classes like this in React, you can call an exposed api method on the component, something like "addClass", which would add it to the component's state, and then rerender it with the class added to `className`

File: src/SearchBar/SearchBar.jsx [hunk 6]
```diff
@@ -54,7 +54,21 @@ class SearchBar extends React.Component {
...
+    window.setTimeout(
+      () => React.findDOMNode(this.refs.input).classList.remove('wiggle'), 1000);
```
Comment (ripleyaffect) [inline src/SearchBar/SearchBar.jsx]: ditto on adding/removing clases

**Comment by ripleyaffect** [comment]
Address comments then :+1:

File: src/AppBar/NavLink.jsx (hunk 1 above)
Comment (learykara) [inline src/AppBar/NavLink.jsx]: yes, as a matter of fact, it will. damnit.

**Pull request merged**